x/gov/client/utils/query.go

```diff
@@ -38,7 +38,15 @@ func (p Proposer) String() string {
 // support configurable pagination.
 func QueryDepositsByTxQuery(clientCtx client.Context, params types.QueryProposalParams) ([]byte, error) {
 	events := []string{
-		fmt.Sprintf("%s.%s='%s'", sdk.EventTypeMessage, sdk.AttributeKeyAction, types.TypeMsgDeposit),
+		fmt.Sprintf(
+			// Msgs `Type()`s are indexed in TM in 2 ways:
+			// - via legacy Msgs (amino or proto), their `Type()` is a custom string,
+			// - via ADR-031 service msgs, their `Type()` is the protobuf FQ method name.
+			// In searching for events, we search for both `Type()`s.
+			"%s.%s='%s' OR %s.%s='%s'",
+			sdk.EventTypeMessage, sdk.AttributeKeyAction, types.TypeMsgDeposit,
+			sdk.EventTypeMessage, sdk.AttributeKeyAction, types.TypeSvcMsgDeposit,
```

Comment:
This is a bugfix that got uncovered when converting CLI to service Msgs.

**Before:**

We only had amino Msgs, their `Type()` was e.g. `"submit_proposal"`. So when searching txs, we were searching `message.action='submit_proposal'`

**Now (since 0.40):**

We have amino Msgs, proto Msgs (`Type()` is same as amino), and service Msgs (`Type()` is proto FQ method name). So for searching txs we need to search for both old and new `Type()`s, e.g.
```
message.action='submit_proposal' OR message.action='/cosmos.gov.v1beta1.Msg/SubmitProposal'
```

this is what I'm doing here.